lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -35,8 +36,8 @@ def deploy_succeeded?
       return false unless @latest_rs
 
       @latest_rs.deploy_succeeded? &&
-      @latest_rs.desired_replicas == desired_replicas && # latest RS fully scaled up
-      @rollout_data["updatedReplicas"].to_i == @rollout_data["replicas"].to_i &&
```

Comment:
In practice it seems like `.status.replicas` (aka `@rollout_data["replicas"].to_i`) pretty much always matches `.spec.replicas`. However, `.spec.replicas` seems to be more strictly the source of truth for desired replica count, so I think we should switch to it for this comparison as well (and the equivalent in the RS class below). A divergence in these values could explain how I still got a 1/200ish false positive rate for a certain previously-more-flakey test on the branch that became v1.7.9.